eole/decoders/transformer.py

```diff
@@ -315,14 +326,15 @@ def forward(self, emb, **kwargs):
             position_embeddings_local = position_embeddings
         decoder_in = kwargs.pop("decoder_in", None)
         image_token_id = kwargs.pop("image_token_id", None)
+        pfxlen = kwargs.pop("pfxlen", None)
```

Comment:
`prefix_len` might be more readable (and coherent with the input data).
Also, this makes me wonder whether we should switch to some "positions-based" logic (like HF) at some point, which might facilitate make some things more generic.
E.g. "causal_positions"/"non_causal_positions" instead of deducing prefix positions or image positions.